rest-service/manager_rest/rest/resources_v3_1/plugins.py

```diff
@@ -63,10 +65,31 @@ def post(self, **kwargs):
         """
         Upload a plugin
         """
+
         visibility = rest_utils.get_visibility_parameter(
             optional=True,
             is_argument=True,
             valid_values=VisibilityState.STATES
         )
         with rest_utils.skip_nested_marshalling():
             return super(Plugins, self).post(visibility=visibility)
```

Comment:
Before this line there should be a `SQL insert` of the plugin with the new PREFIX to verify that the plugin is not being installed.
Verifying that there is not `installing-<plugin_name> is not sufficient since 2 machines can check that and get an OK for it, and immediately after that try to insert the `installing-<plugin_name>`, only one will succeed causing the other to fail which will fail the request.